crate2nix/Cargo.nix

```diff
@@ -2746,9 +2746,31 @@ rec {
           buildInputs = testInputs;
         } ''
           set -ex
-          cd ${crate.src}
+
+          export RUST_BACKTRACE=1
+
+          # recreate a file hierarchy as when running tests with cargo
+
+          # the source for test data
+          ${pkgs.xorg.lndir}/bin/lndir ${crate.src}
+
+          # build outputs
+          testRoot=target/debug
+          mkdir -p $testRoot
+
+          # executables of the crate
+          # we copy to prevent std::env::current_exe() to resolve to a store location
+          cp -r ${crate}/bin/* $testRoot
```

Comment:
This fails when there are no `bin` directory in the compiled crate. We should probably handle that gracefully.

Reply:
This probably also means we are lacking a lib build that is being tested.

Reply:
I replaced this with a for loop for the case where the glob expands to the empty string. I don't understand the second comment.

Reply:
I just meant that since tests did pass on this repository and failed on some of my projects we should probably add a test to this repo where we build a `lib` crate that is being tested.